src/platform/packages/shared/controls/control-group-renderer/src/control_group_renderer.tsx

```diff
@@ -0,0 +1,214 @@
+
+  const lastSavedState$Ref = useRef(new BehaviorSubject<ControlPanelsState>({}));
+
+  /** Creation options management */
+  const { initialState, getEditorConfig } = useInitialControlGroupState(
+    getCreationOptions,
+    lastSavedState$Ref
+  );
+  const input$ = useMemo(() => {
+    if (!initialState) return;
+    return new BehaviorSubject<Partial<ControlGroupRuntimeState>>(initialState ?? {});
+  }, [initialState]);
+
+  const { childrenApi, currentChildState$Ref } = useChildrenApi(initialState, lastSavedState$Ref);
+  const layoutApi = useLayoutApi(initialState, childrenApi, lastSavedState$Ref);
+
+  /** Props management */
+  const searchApi = useSearchApi({
+    filters,
+    query,
+    timeRange,
+  });
+  const propsApi = usePropsApi({ dataLoading, viewMode, compressed });
+
+  const parentApi = useMemo(() => {
+    if (!childrenApi || !layoutApi) return;
+
+    const disabledActionIds$ = new BehaviorSubject<string[] | undefined>(undefined);
+    const reload$ = new Subject<void>();
+
+    return {
+      ...childrenApi,
+      ...layoutApi,
+      ...searchApi,
+      ...propsApi,
+      reload$,
+      getEditorConfig: getEditorConfig.current,
+      disabledActionIds$,
+      setDisabledActionIds: (ids: string[] | undefined) => {
+        disabledActionIds$.next(ids);
+      },
+    };
+  }, [childrenApi, layoutApi, searchApi, propsApi, getEditorConfig]);
+
+  useEffect(() => {
+    if (!parentApi || !input$) return;
+    const currentStateSubscription = combineLatest([
+      currentChildState$Ref.current,
+      parentApi.layout$,
+    ])
+      .pipe(
+        map(([currentChildState, currentLayout]) => {
+          const combinedState: ControlPanelsState = {};
+          Object.keys(currentLayout.controls).forEach((id) => {
+            combinedState[id] = {
+              ...currentChildState[id].rawState,
+              ...currentLayout.controls[id],
+            } as ControlPanelState;
+          });
+          return { initialChildControlState: combinedState };
+        })
+      )
+      .subscribe((currentState) => {
+        input$.next(currentState);
+      });
+
+    return () => {
+      currentStateSubscription.unsubscribe();
+    };
+  }, [input$, currentChildState$Ref, parentApi]);
+
+  useEffect(() => {
+    if (!parentApi || !input$) return;
+
+    /**
+     * the ControlGroupRenderer will render before the children are available and combineCompatibleChildrenApis
+     * will default to the empty value; however, we shouldn't publish this until the value is real
+     */
+    const ignoreWhileLoading = pipe(
+      combineLatestWith(parentApi.childrenLoading$),
+      filter(([, loading]) => !loading),
+      map(([result]) => result)
+    );
+
+    const publicApi = {
+      ...parentApi,
+      esqlVariables$: parentApi.esqlVariables$.pipe(ignoreWhileLoading),
+      filters$: parentApi.appliedFilters$.pipe(ignoreWhileLoading),
+      timeslice$: parentApi.appliedTimeslice$.pipe(ignoreWhileLoading),
```

Comment:
I am removing `applied` to keep the `ControlGroupRendererApi` as similar as possible to the legacy version